crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -56,84 +140,245 @@ impl TryFrom<&JpmorganRouterData<&PaymentsAuthorizeRouterData>> for JpmorganPaym
+                }
+
+                let capture_method =
+                    map_capture_method(item.router_data.request.capture_method.unwrap_or_default());
+
+                let merchant_software = JpmorganMerchantSoftware {
+                    company_name: String::from("JPMC"),
+                    product_name: String::from("Hyperswitch"),
+                };
+
+                let merchant = JpmorganMerchant { merchant_software };
+
+                let expiry: Expiry = Expiry {
+                    month: req_card.card_exp_month.clone(),
+                    year: req_card.get_expiry_year_4_digit(),
+                };
+
+                let account_number = Secret::new(req_card.card_number.to_string());
+
                 let card = JpmorganCard {
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
+                    account_number,
+                    expiry,
                 };
+
+                let payment_method_type = JpmorganPaymentMethodType { card };
+
                 Ok(Self {
-                    amount: item.amount.clone(),
-                    card,
+                    capture_method: capture_method?,
+                    currency: item.router_data.request.currency,
+                    amount: item.amount,
+                    merchant,
+                    payment_method_type,
                 })
             }
-            _ => Err(errors::ConnectorError::NotImplemented("Payment method".to_string()).into()),
+            PaymentMethodData::CardDetailsForNetworkTransactionId(_)
+            | PaymentMethodData::CardRedirect(_)
+            | PaymentMethodData::Wallet(_)
+            | PaymentMethodData::PayLater(_)
+            | PaymentMethodData::BankRedirect(_)
+            | PaymentMethodData::BankDebit(_)
+            | PaymentMethodData::BankTransfer(_)
+            | PaymentMethodData::Crypto(_)
+            | PaymentMethodData::MandatePayment
+            | PaymentMethodData::Reward
+            | PaymentMethodData::RealTimePayment(_)
+            | PaymentMethodData::MobilePayment(_)
+            | PaymentMethodData::Upi(_)
+            | PaymentMethodData::Voucher(_)
+            | PaymentMethodData::GiftCard(_)
+            | PaymentMethodData::OpenBanking(_)
+            | PaymentMethodData::CardToken(_)
+            | PaymentMethodData::NetworkToken(_) => Err(errors::ConnectorError::NotImplemented(
+                get_unimplemented_payment_method_error_message("jpmorgan"),
+            )
+            .into()),
         }
     }
 }
-
-//TODO: Fill the struct with respective fields
-// Auth Struct
+#[derive(Debug)]
 pub struct JpmorganAuthType {
-    pub(super) api_key: Secret<String>,
+    pub(super) _api_key: Secret<String>,
+    pub(super) _key1: Secret<String>,
```

Comment:
Can we remove the underscore prefix if we are using this struct?

If we are not using this struct, why are we not using it?

Reply:
removed the struct

Reply:
These keys are unused because JPMorgan only requires an access token. The access token is obtained during the token flow using the JPMorgan API keys, which are provided through the `AccessTokenRequestData`.